sale_channel_mirakl/tests/common.py

```diff
@@ -0,0 +1,577 @@
+import csv
+from base64 import b64decode
+from io import StringIO
+
+from odoo import Command
+from odoo.tests import common
+
+
+class SetUpMiraklBase(common.TransactionCase):
+    def _read_filename(self, filename):
+        csv_file = self.env["ir.attachment"].search([("name", "=", filename)], limit=1)
+        datas = b64decode(csv_file.datas)
+        reader = csv.DictReader(StringIO(datas.decode()), delimiter=";")
+        return reader
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.sale_channel = cls.env["sale.channel"].create(
+            {
+                "name": "Super sale channel",
+                "channel_type": "mirakl",
+            }
+        )
+
+        cls.sale_channel_mirakl = cls.env["sale.channel.mirakl"]
+
+        cls.mirakl_sc_for_product = cls.sale_channel_mirakl.create(
+            {
+                "channel_id": cls.sale_channel.id,
+                "location": "http://anyurl.com",
+                "api_key": "azerty",
+                "offer_filename": "offers.csv",
+                "shop_id": "11",
+                "data_to_export": "product",
+            }
+        )
+
+        cls.mirakl_sc_for_offer = cls.sale_channel_mirakl.create(
+            {
+                "channel_id": cls.sale_channel.id,
+                "location": "http://anyurl.com",
+                "api_key": "azerty",
+                "offer_filename": "offers.csv",
+                "shop_id": "25",
+                "data_to_export": "offer",
+            }
+        )
+
+        cls.mirakl_sc_for_catalog = cls.sale_channel_mirakl.create(
+            {
+                "channel_id": cls.sale_channel.id,
+                "location": "http://anyurl.com",
+                "api_key": "azerty",
+                "offer_filename": "offers.csv",
+                "shop_id": "25",
+                "data_to_export": "both",
+            }
+        )
+
+        cls.product1 = cls.env.ref("product.product_product_9")
+        cls.product2 = cls.env.ref("product.product_product_11")
+
+        cls.product1.write(
+            {
+                "default_code": "29491209",
+                "mirakl_id": "29491209",
```

Comment:
IMO the `mirakl_id` should be defined on the relation between the product and the mirakl sale channel.

Otherwise, that means a product will have the same `mirakl_id` into every channel.

I think there is something to dig here!